peer_review_data/main.py

```diff
@@ -99,103 +96,116 @@ def saveAssessmentsAndComments(
     for canvasAssessment in [CanvasAssessment(a) for a in
                              canvasAssessments]:
         if not canvasAssessment.isPeerReview:
-            LOGGER.info('Skipping non-peer-review assessment '
-                        f'({canvasAssessment.id}).')
-            continue
+            LOGGER.warning(f'Assessment ({canvasAssessment.id}) '
+                           'is NOT a peer-review.')
 
-        assessment: models.Assessment = None
+        assessment: Optional[models.Assessment] = None
         try:
             assessment = \
                 models.Assessment.fromCanvasAssessment(canvasAssessment)
+
+            if assessment is None:
+                continue
+
             LOGGER.debug(f'Saving {assessment}…')
             assessment.save()
         except Exception as e:
-            # XXX: Catches assessments referring to non-existent submissions!
-            LOGGER.warning(f'Error saving Assessment '
-                           f'({canvasAssessment.id}): {e}')
+            # XXX: Catch assessment referring to missing assessor or submission
+            LOGGER.warning(f'Error saving Assessment ({canvasAssessment.id}; '
+                           f'Submission: {canvasAssessment.submissionId}; '
+                           f'Assessor: {canvasAssessment.assessorId}): {e}')
             LOGGER.debug('Assessment data: ' + json.dumps(
```

Comment:
Not sure of all the scenarios here, but this seems like a classic LBYL vs. EAFP scenario. Not sure which approach to go with each scenario, but these seem to be happening with some regularlity, so I'd probably think about using more LBYL. https://realpython.com/python-lbyl-vs-eafp/#eafp-vs-lbyl-with-examples

Reply:
Thanks for the suggestion.  I had thought about checking data before trying to save it, but that would be one or more extra steps.  The Django code that saves the model will do its checks even if this code checks ahead of time.  So it seems to me that the less code written to check, the less code that could have bugs.

I agree that I should use more specific exceptions than `Exception`.  I'll look into that and perhaps with parsing the exception messages for data that explain the problem, if the exceptions don't already contain the data in some other field.

Reply:
I think this is good as it is.  If anyone thinks this needs to be addressed, let's open a new issue for it.

Reply:
Replaced `Exception` with more specific exceptions in b6c6c30.